b2share/modules/main/restful.py

```diff
@@ -0,0 +1,230 @@
+        by current user. The default state of record status is 'draft'
+        """
+        return None
+
+
+class Record(Resource):
+    def get(self, record_id, **kwargs):
+        """
+        Returns the requested record
+        """
+        return None
+
+    def patch(self, record_id, **kwargs):
+        """
+        Edits the Record and submit it.
+        """
+        return None
+
+    def delete(self, record_id, **kwargs):
+        """
+        Deletes the record:
+            - if the state of record is draft: it will be deleted
+            - if the state is released: it will be marked as deleted.
+                Only available to superadmins
+        """
+        return None
+
+
+# ----------------------------------------------- FILES
+
+
+class FileList(Resource):
+    def get(self, record_id, **kwargs):
+        """
+        Retrieve list of files in a record.
+        """
+        return None
+
+    def post(self, record_id, **kwargs):
+        """
+        Add a new file to the record. The record state should be 'draft'
+        """
+        return None
+
+
+class FileContainer(Resource):
+    def get(self, record_id, file_id, **kwargs):
+        """
+        Get a file metadata
+        """
+        return None
+
+    def patch(self, record_id, file_id, **kwargs):
+        """
+        Rename the file or container
+        """
+        return None
+
+    def delete(self, record_id, file_id, **kwargs):
+        """
+        Delete a file/container (only possible for draft records).
+        If it is a container it deletes recursively its children.
+        """
+        return None
+
+
+class File(Resource):
+    def get(self, record_id, file_id, **kwargs):
+        """
+        Get a file's content
+        """
+        return None
+
+
+# ----------------------------------------------- REFERENCES
+
+
+class RecordReferenceList(Resource):
+    def get(self, record_id, **kwargs):
+        """
+        Returns the record's references
+        """
+        return None
+
+    def post(self, record_id, **kwargs):
+        """
+        Create a new reference
+        """
+        return None
+
+
+class RecordReference(Resource):
+    def get(self, record_id, ref_id, **kwargs):
+        """
+        Returns a Reference with the specified id
+        """
+        return None
+
+    def delete(self, record_id, ref_id, **kwargs):
+        """
+        Deletes a reference
+        """
+        return None
+
+
+# ----------------------------------------------- USERS
+
+
+class UserList(Resource):
+    def get(self, **kwargs):
+        """
+        Search for users.
+        """
+        return None
+
+
+class User(Resource):
+    def get(self, user_id, **kwargs):
+        """
+        Get a user profile information
+        """
+        return None
+
+
+def setup_app(app, api):
+    api.add_resource(CommunityList, '/api/communities')
+    api.add_resource(CommunityRes, '/api/communities/create')
```

Comment:
We've decided not to use verbs in the URLs, so this one must go. The same functionality can be achieved by a POST in CommunityList.